_posts/2019-01-05-DQN.markdown

```diff
@@ -0,0 +1,294 @@
+---
+layout: post
+title:  "Deep Q-Networks"
+date:   2019-01-05 10:10:04 +0530
+categories: rl
+author: Akash Nair
+---
+
+In this article, we will learn about Deep Q-Networks and a basic implementation of it.
+
+## **Recap**
+
+## Q-Learning and some Formal Definitions 
+
+### What is a Policy?
+
+We can define **policy** as a map from states to actions, its the objective in reinforcement learning to find the optimal policy, and this optimal policy can be derived from a Q function.
+
+### Return / Future Total Reward
+
+We define return as the sum of all immediate rewards on a particular episode, ie, the sum of rewards up until the episode finish.
+
+### Future Discounted Reward
+
+We add to our total reward a parameter called gamma ( $$ \gamma $$ ). If $$ \gamma $$ = 0, all future rewards will be discarded otherwise they will be considered. 
+We can interpret this parameter as in terms of how much we value our future rewards compared to our current reward.
+
+### What is an Episode?
+
+Our environment is based on an MDP (Markov Decision Process) which means that our state encodes everything needed to take a decision by defining an episode as a finite set of states, actions and rewards.
+Here, we get a reward taking the action from a state and an episode always finishes on an end/final state.
+
+### Q function
+
+We define the function $$ Q_{(s,a)} $$ as maximum expected return that we can get if we take an action $$ a $$ on state $$ s $$, and continue optimally from that point until the end of the episode which means that we continue choosing actions from the policy derived from Q.
```

Comment:
"continue optimally" - That's incorrect. The state action function is defined over a policy so, the q-value is the expected return from that state-action pair following the current policy. So remove the optimally part